applications/dashboard/settings/class.hooks.php

```diff
@@ -266,6 +266,23 @@ public function gdn_dispatcher_appStartup_handler($Sender) {
     }
 
     /**
+     * @param Gdn_Dispatcher $sender
+     */
+    public function gdn_dispatcher_sendHeaders_handler($sender) {
+        $csrfToken = Gdn::request()->post(
+            Gdn_Session::CSRF_NAME,
+            Gdn::request()->get(
+                Gdn_Session::CSRF_NAME,
+                Gdn::request()->getValueFrom(Gdn_Request::INPUT_SERVER, 'HTTP_X_CSRF_TOKEN')
+            )
+        );
+
+        if ($csrfToken && Gdn::session()->isValid() && !Gdn::session()->validateTransientKey($csrfToken)) {
```

Comment:
My worry here is that someone posting a fake TK will get back the proper TK in the response.